src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,159 @@
+# coding=utf-8
+# Copyright 2023 The Suno AI Authors and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for Bark
+"""
+from typing import Optional
+
+import numpy as np
+
+from ...processing_utils import ProcessorMixin
+from ...utils.hub import cached_file
+from ..auto import AutoTokenizer
+
+
+class BarkProcessor(ProcessorMixin):
+    r"""
+    Constructs a Bark processor which wraps a Bark voice preset and a text tokenizer into a single processor.
+
+    Args:
+        tokenizer ([`PreTrainedTokenizer`], *optional*):
+            An instance of [`PreTrainedTokenizer`]. The tokenizer is optional. By default, it will instantiated as
+            "bert-base-multilingual-cased".
+        repo_id (`str`, *optional*, defaults to ""):
+            This can be either:
+
+            - a string, the *model id* of a model repo on huggingface.co.
+            - a path to a *directory* potentially containing the file.
+        subfolder (`str`, *optional*, defaults to `""`):
+            In case the relevant files are located inside a subfolder of the model repo on huggingface.co, you can
+            specify the folder name here.
+    """
+    tokenizer_class = "AutoTokenizer"
+    attributes = ["tokenizer"]
+
+    preset_shape = {
+        "semantic_prompt": 1,
+        "coarse_prompt": 2,
+        "fine_prompt": 2,
+    }
+
+    def __init__(self, tokenizer: Optional[AutoTokenizer] = None, repo_id="", subfolder="", **kwargs):
+        if tokenizer is None:
+            tokenizer = AutoTokenizer.from_pretrained("bert-base-multilingual-cased")
+        super().__init__(tokenizer)
+
+        self.repo_id = repo_id
+        self.subfolder = subfolder
+
+    @classmethod
+    def from_pretrained(
+        cls, pretrained_tokenizer_name_or_path, repo_id=None, subfolder="speaker_embeddings", **kwargs
```

Comment:
IMO `from_pretrained` should only have one `pretrained_model_name_or_path` argument - this is the path to the tokenizer, and also the `repo_id`.

Reply:
So should I load `bert-base-multilingual-cased` to the `bark-small` and `bark-large` repos? I feel like it leaves the impression that `bark` uses a specific tokenizer instead of an already widely used tokenizer.

Reply:
For example, Whisper uses the same tokenizer as GPT2 - what we do here in the conversion script is load the GPT2 tokenizer from the original GPT2 repo, then save it in the new Whisper repo. Now, when a user wants to load the Whisper processor / tokenizer from pre-trained, they can do so directly from the new Whisper repo

Reply:
So the new Whisper repo is 'complete' in this sense and contains all the config / weights that someone would need to run the Whisper model for themselves